asset-transfer-abac/README.md

```diff
@@ -0,0 +1,176 @@
+# Attribute based access control
+
+The `asset-transfer-abac` sample demonstrates the use of Attribute-based access control within the context of a simple asset transfer scenario. The sample also uses authorization based on individual client identities to allow the users that interact with the network to own assets on the blockchain ledger.
+
+Attribute-Based Access Control (ABAC) refers to the ability to restrict access to certain functionality within a smart contract based on the attributes within a users certificate. For example, you may want certain functions within a smart contract to be used only by application administrators. ABAC allows organizations to provide elevated access to certain users without requiring those users be administrators of the network. For more information, see [Attribute-based access control](https://hyperledger-fabric-ca.readthedocs.io/en/latest/users-guide.html#attribute-based-access-control) in the Fabric CA users guide.
+
+The `asset-transfer-abac` smart contract allows you to create assets that can be updated or transferred by the asset owner. However, the ability to create or remove assets from the ledger is restricted to identities with the `abac.creator=true` attribute. The identity that creates the asset is assigned as the asset owner. Only the owner can transfer the asset to a new owner or update the asset properties. In the course of the tutorial, we will use the Fabric CA client to create identities with the attribute required to create a new asset. We will then transfer the asset to another identity and demonstrate how the `GetID()` function can be used to enforce asset ownership. We will then delete the asset using another identity with the `abac.creator=true` attribute. Both Attribute based access control and the `GetID()` function are provided by the [Client Identity Chaincode Library](https://github.com/hyperledger/fabric-chaincode-go/blob/master/pkg/cid/README.md).
+
+## Start the network and deploy the smart contract
+
+We can use the Fabric test network to deploy and interact with the `asset-transfer-abac` smart contract. Run the following command to change into the test network directory and bring down any running nodes:
+```
+cd fabric-samples/test-network
+./network.sh down
+```
+
+Run the following command to deploy the test network using Certificate Authorities:
+```
+./network.sh up createChannel -ca
+```
+
+You can then use the test network script to deploy the `asset-transfer-abac` smart contract to a channel on the network:
+```
+./network.sh deployCC -ccn abac -ccp ../asset-transfer-abac/chaincode-go/
```

Comment:
also pass -ccl based on the changes to be more explicit.

Reply:
done.